### ISSUE TITLE
[build] Permissions not being removed during build

### ISSUE THREAD

**Issue by BrodaNoel** [expo/eas-cli#1077]
### Summary

I have the feeling that something is broken in `eas build` (or, as usual, I'm missing something)

Take a look at this issue: https://github.com/expo/expo/issues/14525

We are patching some changes in `node_modules` in the `post-install`, in order to remove some Permissions that are unnecessary added, but then, in the build (made with `eas build`), the permissions are still there. Like... `eas build` is ignoring the changes we are applying.

I still have some doubts if we are doing everything well there. For example:
- We are applying the changes in the `build` folder instead of in the `src` folder. Not sure if this is correct. Not sure if `eas build` builds everything from scratch, or if it uses the `build` folder.
- I'm not sure if the correct hook to apply the patch is `post-install` or if it's other.
- We are not clearing cache (in case it is necessary)

Still, feels like we are missing something, or it's a possible bug in `eas build`

### Managed or bare?

Managed

### Environment

  expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 11.6.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.18.2 - ~/.nvm/versions/node/v14.18.2/bin/node
      Yarn: 1.22.18 - ~/.yarn/bin/yarn
      npm: 8.3.2 - ~/.nvm/versions/node/v14.18.2/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
      Android SDK:
        API Levels: 27, 29, 30
        Build Tools: 23.0.1, 23.0.3, 25.0.3, 26.0.1, 27.0.3, 28.0.3, 29.0.2
        System Images: android-29 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7784292
      Xcode: 12.5.1/12E507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.6
      react: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
    npmGlobalPackages:
      eas-cli: 0.51.0
      expo-cli: 5.3.2
    Expo Workflow: managed

### Error output

No error

### Reproducible demo or steps to reproduce from a blank project

Here it should be enough with adding `expo-image-picker` to the dependencies (I'm using `"expo-image-picker": "~12.0.1"`).

```
import * as ImagePicker from 'expo-image-picker';
```

Then follow this instructions: https://github.com/expo/expo/issues/14525#issuecomment-986719649 in order to apply the patch to the `build` files, and then `eas build`

**Comment by wkozyra95** [comment]
patch-package definitely works on eas, if you have some problems with getting that working you can debug it with `eas build:inspect --stage=pre-build` command and inspect the resulting build directory if everything was applied correctly

As for the instruction, I think use of `eas-build-post-install` is not correct, it should be done in `postinstall`(there is no such thing as `post-install`).

**Comment by BrodaNoel** [comment]
@wkozyra95 what do you mean with there is no post-install? I can see it in the documentation

![image](https://user-images.githubusercontent.com/1837969/164453999-e0db5105-3b49-4bba-b664-a3d70f8d59e4.png)

**Comment by wkozyra95** [comment]
There is `eas-build-post-install` there is no `post-install`, either way, `eas-build-post-install` is the wrong place to apply that patch

**Comment by BrodaNoel** [comment]
Thanks! Do you have any idea which is the correct place to apply the patch?

**Comment by wkozyra95** [comment]
> it should be done in `postinstall`

https://docs.npmjs.com/cli/v8/using-npm/scripts

**Comment by BrodaNoel** [comment]
@wkozyra95 confirmed.

1. I ran `eas build:inspect --stage=pre-build -p android -o ./prebuild --verbose`, and I could saw the permissions were present in `android/app/src/main/AndroidManifest.xml`
2. I changed `eas-build-post-install` with `postinstall`
3. I ran `eas build:inspect --stage=pre-build -p android -o ./prebuild --verbose`, and now permissions are not there anymore.

Feel free to close this issue if you consider there is no documentation to change in EAS-CLI.

I will push the new build to Google Play and see if the unnecessary permission is gone, and then update the original bug in `/expo/expo`
